addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/spider/ExtensionOpenApiSpider.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.openapi.spider;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.addon.spider.ExtensionSpider2;
+import org.zaproxy.addon.spider.parser.SpiderParser;
+
+public class ExtensionOpenApiSpider extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionOpenApiSpider";
+    private static final Logger LOG = LogManager.getLogger(ExtensionOpenApiSpider.class);
+    private SpiderParser customSpider;
+
+    public ExtensionOpenApiSpider() {
+        super(NAME);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        /* Custom spider is added in order to explore Open API definitions. */
+        ExtensionSpider2 spider =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionSpider2.class);
+        customSpider = new OpenApiSpider();
+        if (spider != null) {
```

Comment:
This should not be needed, the extension should depend on `ExtensionSpider2` (and `ExtensionOpenApi`).